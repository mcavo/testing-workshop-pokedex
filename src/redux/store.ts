import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import Reactotron from 'reactotron-react-native';
import thunk from 'redux-thunk';
import { fetchMiddleware, configureMergeState } from 'redux-recompose';
import { ImmutableObject } from 'seamless-immutable';
import { IState } from '@interfaces/reduxInterfaces';

import auth from './auth/reducer';
import pokemon from './pokemon/reducer';

configureMergeState((state: ImmutableObject<IState>, diff: IState) => state.merge(diff));

const reducers = combineReducers({
  auth,
  pokemon
});

const middlewares = [];
const enhancers = [];

/* ------------- Thunk Middleware ------------- */
middlewares.push(thunk);

/* ------------- Redux-Recompose Middleware ------------- */
middlewares.push(fetchMiddleware);

/* ------------- Assemble Middleware ------------- */
enhancers.push(applyMiddleware(...middlewares));

if (__DEV__ && Reactotron.createEnhancer) enhancers.push(Reactotron.createEnhancer(true));

// In DEV mode, we'll create the store through Reactotron
const store = createStore(reducers, compose(...enhancers));

if (__DEV__ && Reactotron.setReduxStore) Reactotron.setReduxStore(store);

export default store;
