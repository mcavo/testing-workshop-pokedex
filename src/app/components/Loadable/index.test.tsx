import React from 'react';
import { render } from '@testsUtils/redux';
import { View } from 'react-native';

import withLoadable from './index';

describe('withLoadable', () => {
  const LoadableView = withLoadable(() => true)(View);
  test('withLoadable Snapshot', () => {
    const loadableView = render(<LoadableView />).toJSON();
    expect(loadableView).toMatchSnapshot();
  });
});
