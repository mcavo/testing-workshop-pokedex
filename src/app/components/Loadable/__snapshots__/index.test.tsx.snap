// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoadable withLoadable Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color={Object {}}
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
